service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderController.java

```diff
@@ -0,0 +1,91 @@
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.order.generated.GeneratedOrder;
+import uk.gov.hmcts.reform.fpl.service.RemoveOrderService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.removeTemporaryFields;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.getDynamicListValueCode;
+
+@Api
+@RestController
+@RequestMapping("/callback/remove-order")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class RemoveOrderController {
+    private final ObjectMapper mapper;
+    private final RemoveOrderService service;
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest request) {
+        Map<String, Object> data = request.getCaseDetails().getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        data.put("removableOrderList", service.buildDynamicListOfOrders(caseData.getOrderCollection()));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest request) {
+        Map<String, Object> data = request.getCaseDetails().getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        // When dynamic lists are fixed this can be moved into the below method
+        UUID id = getDynamicListValueCode(caseData.getRemovableOrderList(), mapper);
+
+        data.putAll(service.populateSelectedOrderFields(caseData.getOrderCollection(), id));
+
+        // Can be removed once dynamic lists are fixed
+        data.put("removableOrderList", service.buildDynamicListOfOrders(caseData.getOrderCollection(), id));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data)
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest request) {
+        CaseDetails caseDetails = request.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        List<Element<GeneratedOrder>> orders = caseData.getOrderCollection();
+        List<Element<GeneratedOrder>> hiddenOrders = caseData.getHiddenOrders();
+
+        service.hideOrder(orders, hiddenOrders, caseData.getRemovableOrderList(), caseData.getReasonToRemoveOrder());
+
+        data.put("orderCollection", orders);
+        data.put("hiddenOrders", hiddenOrders);
+        removeTemporaryFields(
+            caseDetails,
+            "removableOrderList",
```

Comment:
code smell complaining about "removableOrderList" being referenced like this three times.. is it something we want to put at top of class like as static variable or do we want to ignore the code smell?

Reply:
moved as a static variable